pkg/controller/workload/kubernetes/resource/resource.go

```diff
@@ -266,9 +266,7 @@ func (c *unstructuredClient) sync(ctx context.Context, template *unstructured.Un
 	}
 
 	rs := getRemoteStatus(remote)
-
-	// TODO(negz): Update to resource.APIPatchingApplicator.
-	return rs, errors.Wrap(resource.Apply(ctx, c.kube, template), errSyncResource) // nolint:staticcheck
+	return rs, errors.Wrap(resource.NewAPIPatchingApplicator(c.kube).Apply(ctx, template), errSyncResource)
```

Comment:
@negz went ahead and made this update because it looked like it is functionally the same.